design-documents/catalog-permissions.md

```diff
@@ -0,0 +1,293 @@
+# Catalog permissions
+
+## Problem statement
+
+The goal of this proposal is to define how catalog permissions will be synchronized to the storefront. It also proposes to replace customer groups and website, used to represent product visibility with catalogs (the exact name should be defined).
+
+## Overview of category permissions
+
+Visibility for categories and products determined by the following options
+
+* Website assignment
+* Catalog permissions
+
+Catalog permission settings looks like this and applied on the category level.
+
+![Catalog permissions management](./catalog-permissions/catalog-permissions-management.png)
+
+Category permissions set on the website and category level and applied to products in the category.
+
+Below is the example of how data looks like for 2 customer groups.
+
+```json
+{
+	"permissions": {
+		"1": {
+			"is_visible": "true",
+			"allow_display_prices": "true",
+			"allow_add_to_cart": "true",
+		},
+		"2": {
+			"is_visible": "true",
+			"allow_display_prices": "true",
+			"allow_add_to_cart": "true",
+		}
+	}
+}
+```
+
+## Using catalogs to represent visibility
+
+Catalog are going to be used to represent visibility of the product. To represent permissions and websites with catalogs we will have to create catalog for website and customer group (for which we have catalog permissions configured) combinations.
+
+More information on catalogs [here](https://wiki.corp.magento.com/pages/viewpage.action?spaceKey=ARCH&title=Organization+Structure%2C+Magento+Instance+Registration+and+Scope+Isolation).
```

Comment:
Can you put here what exactly we need from "catalog" and what is it (for those who can't open resource)
Can we implement this proposal without implementation mentioned "catalog"?